pyomo/contrib/pynumero/src/hsl_interface/Makefile

```diff
@@ -0,0 +1,54 @@
+rootdir:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
```

Comment:
I am curious why using makefiles and not include this in the cmake project for pynumero to build a separate library for hsl

Reply:
This should be incorporated into the pynumero cmake project. The makefiles are basically just placeholders until I can learn enough cmake to incorporate these libraries into the existing project. 

Also, at some point I was planning to include targets for the dependency libraries/objects, e.g. `libmetis.a`, `ma27d.o`, etc, as some of these need special rules (adding -fPIC) if compiled individually and not as part of `libcoinhsl.so`. I thought this would be easier to do with makefiles than with cmake. Not sure if I'm still planning to do this, and regardless of how the dependencies are compiled, I'm still planning to have the pynumero libraries be part of the cmake project.